### PR TITLE
feat(core): Move execution permission checks earlier in the lifecycle

### PR DESCRIPTION
**Pull request by netroy**
This change move the execution permission check before the execution start. 
This fixes issues like [this](https://n8nio.sentry.io/issues/4965073076) and [this](https://n8nio.sentry.io/issues/4965073078).

Also, deleted some own-mode related code.

## Review / Merge checklist
- [x] PR title and summary are descriptive
- [ ] Tests included

### CODE REVIEW

File: packages/workflow/src/errors/credential-access-error.ts
Comment (krynble) [inline packages/workflow/src/errors/credential-access-error.ts:1]: Not sure this is in the scope of this PR

**Review by krynble — DISMISSED** [review]
LGTM, adding `workflowExecuteBefore` ensures we run all lifecycle hooks for the workflow. There are a couple changes I don't think are in the scope of this PR but won't hurt.

File: packages/workflow/src/errors/credential-access-error.ts
Comment (netroy) [reply packages/workflow/src/errors/credential-access-error.ts:1]: moved to https://github.com/n8n-io/n8n/pull/8574

**Review by krynble — DISMISSED** [review]

**Review by krynble — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.30.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.30.0)

**Pull request merged**